control-plane/agents/common/src/v0/msg_translation.rs

```diff
@@ -117,6 +118,27 @@ impl RpcToMessageBus for rpc::Nexus {
             children: self.children.iter().map(|c| c.to_mbus()).collect(),
             device_uri: self.device_uri.clone(),
             rebuilds: self.rebuilds,
+            share: uri_to_protocol(&self.device_uri),
+        }
+    }
+}
+
+/// Convert a device URI to a share Protocol
+/// Uses the URI scheme to determine the protocol
+/// Temporary WA until the share is added to the mayastor RPC
+fn uri_to_protocol(uri: &str) -> mbus::Protocol {
```

Comment:
Tryfrom ?

Reply:
Added it to the api crate where the Protocol is defined.